src/services/services.ts

```diff
@@ -1086,6 +1088,12 @@ namespace ts {
         newText: string;
     }
 
+    export interface TextInsertion {
+        newText: string;
+        /** The position in newText the caret should point to after the insertion. */
+        caretOffset: number;

```

Comment:
> YAGNI

YAGNI implies I'm trying to cover us for when/if we adopt that model. I'm suggesting that we just adopt that model. Maybe @CyrusNajmabadi can weigh in.